src/cpp/core/r_util/RProjectFile.cpp

```diff
@@ -324,10 +326,21 @@ Error findProjectFile(FilePath filePath,
 
    if (!filePath.isDirectory())
       filePath = filePath.parent();
+   
+   // also screen parent of anchor path
+   FilePath anchorParentPath = anchorPath.parent();
 
    // no .Rproj file found; scan parent directories
    for (; filePath.exists(); filePath = filePath.parent())
    {
+      // bail if we've hit our anchor
+      if (filePath == anchorPath)
+         return fileNotFoundError(ERROR_LOCATION);
+      
+      // bail if we're no longer within the anchor's parent path
+      if (!filePath.isWithin(anchorParentPath))
```

Comment:
Not sure I understand this bit. How could we reach a file path that isn't within the anchor's parent without first traversing past the anchor itself?

If the intent is to bail out if the filePath is not within the original anchor, it seems like we could do that just once instead of inside the loop (let me know if I'm misreading!).

Reply:
I was aiming to capture situations where we might have something like the following:

```
anchor: /home/users/me
file: /home/users/you/project/file.R
```

In the current scheme, we'd start testing the directories `/home/users/you/project`, `/home/users/you`, `/home/users` and `/home` for project files -- I wanted to ensure that we would stop once we hit `/home/users` in the above example. I'll see if I can express that more clearly.